api/app/models/event.rb

```diff
@@ -106,7 +106,7 @@ def to_s
 
   # rubocop:disable Rails/SkipsModelValidations
   def touch_project!
-    project.touch
+    project&.touch unless project&.new_record?
```

Comment:
Didn't test this, but seems like it might be nicer to read if it was like

```
project.touch if project.persisted?
```

or something